runtime/runtime/src/verifier.rs

```diff
@@ -57,6 +57,19 @@ pub fn validate_transaction(
         return Err(InvalidTxError::InvalidSignature.into());
     }
 
+    #[cfg(feature = "protocol_feature_tx_size_limit")]
```

Comment:
This check should only be done once. There is no reason to check transactions for which you already checked. For example, inside `prepare_transactions` it doesn't make sense to check again because this check should have been done when the transaction is received. To further reduce the overhead, you could change `NetworkClientMessage::Transaction` to include the transaction size. It probably even makes sense to implement custom borsh deserializer that would fail when the a larger-than-allowed transaction arrives on the network (this can potentially be done in a separate PR)

Reply:
It seems that at least one serialization is already happening inside `SignedTransaction::new`, to save `Transaction` hash. So I saved `size` there in the same fashion and started to use it, which shouldn't give significant overhead.

Reply:
Yeah that makes sense